File: primesieve.cpp
#include<iostream>

using namespace std;

int main()
{
	int n, q, i;
    cin >> n >> q;

	int t[q];

    for (i=0; i<q; i++) {
        cin >> t[i];
    }

	int c = 0;
	bool prime[n+1];
	
	for (i=0; i<n+1; i++) {
		prime[i] = true;
	}

	for (int p=2; p*p<=n; p++)
	{
		// If prime[p] is not changed, then it is a prime
		if (prime[p] == true)
		{
			// Update all multiples of p
			for (int i=p*2; i<=n; i += p)
				prime[i] = false;
		}
	}

	for (int p = 2; p <= n; p++)
        if (prime[p]) c += 1;

	cout << c << endl;

    for (int i=0; i<q; i++) {
		// cout << t[i] << endl;
		if (t[i] == 1) cout << 0 << endl;
		else if (prime[t[i]] == true) {
			cout << 1 << endl;
		}
		else cout << 0 << endl;
	}

	return 0;
}

File: primesieve.py

def SieveOfEratosthenes(num):
    prime = [True for i in range(num+1)]
    # boolean array
    c = 0
    p = 2
    while (p * p <= num):
  
        # If prime[p] is not
        # changed, then it is a prime
        if (prime[p] == True):
            # Updating all multiples of p
            for i in range(p * p, num+1, p):
                prime[i] = False

        if (prime[p]): c += 1
        p += 1
  
    return prime, c


if __name__ == '__main__':
    n, q = map(int, input().split(' '))
    sieve, c = SieveOfEratosthenes(n)
    print(c)
    print(sieve.count(True)-2)
    for _ in range(q):
        i = int(input())
        if i == 1:
            print(0)
        else:
            print(1) if sieve[i] else print(0)
